### PR TITLE
Add installation script

### PR DESCRIPTION
**Pull request by briandowns**
Add an installation script based off of the k3s install script and account for installing an rke2 and etcd user, updating permissions, etc. Side note, bash-isms have been removed for the closest I can get the script to posix compatible. I tested in dash and busybox so bash no longer is a requirement.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,877 @@
+#!/bin/sh
+
+set -e
+
+if [ "${DEBUG}" = 1 ]; then 
+    set -x
+fi
+
+# Usage:
+#   curl ... | ENV_VAR=... sh -
```
Comment (galal-hussein) [inline install.sh:10]: Wondering if we should remove any curl mention, in this script since its not uploaded yet

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+#       or
+#   ENV_VAR=... ./install.sh
+#
+# Example:
+#   Installing a server without traefik:
+#     curl ... | INSTALL_RKE2_EXEC="--disable=traefik" sh -
+#   Installing an agent to point at a server:
+#     curl ... | RKE2_TOKEN=xxx RKE2_URL=https://server-url:6443 sh -
```
Comment (galal-hussein) [inline install.sh:18]: so in rke2 the supervisor port is set to 9345 by default, so this command will fail if its using 6443

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,877 @@
...
+#
+#   - INSTALL_RKE2_ETCD_USER
+#     Create a user 'etcd'. If this value is set, the installation
+#     will chown the etcd data-dir to this user and update the etcd
+#     pod manifest.
+
+BASE_DIR="/var/lib/rancher/rke2"
+INSTALL_PATH="/usr/local/bin"
+GITHUB_URL=https://github.com/rancher/rke2/releases
+STORAGE_URL=https://storage.googleapis.com/rke2-ci-builds
```
Comment (galal-hussein) [inline install.sh:88]: There is no ci builds storage sat up for rke2 yet so this wont work properly

File: install.sh (hunk 1 above)
Comment (briandowns) [reply install.sh:10]: I'll remove it and add it back when that feature is possible.

File: install.sh (hunk 3 above)
Comment (briandowns) [reply install.sh:82]: I left this in, in the hopes we have that soon. I can remove.

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,877 @@
...
+        arm64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        aarch64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        arm*)
+            ARCH=arm
```
Comment (galal-hussein) [inline install.sh:289]: there is no support for arm yet for rke2

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,877 @@
...
+        rm -rf "${TMP_DIR}"
+        exit $code
+    }
+    trap cleanup INT EXIT
+}
+
+# --- use desired rke2 version if defined or find version from channel ---
+get_release_version() {
+    if [ -n "${INSTALL_RKE2_COMMIT}" ]; then
+        VERSION_RKE2="commit ${INSTALL_RKE2_COMMIT}"
```
Comment (galal-hussein) [inline install.sh:332]: I guess we should either set up the same commit storage space for rke2 like k3s or skip this option

File: install.sh (hunk 4 above)
Comment (briandowns) [reply install.sh:289]: Was leaving that as well as that's on the way.

File: install.sh (hunk 5 above)
Comment (briandowns) [reply install.sh:332]: I agree. :D

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,871 @@
...
+
+# re-evaluate args to include env command
+eval set -- $(escape "${INSTALL_RKE2_EXEC}") $(quote "$@")
+
+# main
+{
+    if [ ! -z "${INSTALL_RKE2_USER}" ]; then
+        mkdir -p "${BASE_DIR}"
+        create_user "${INSTALL_RKE2_USER}" "RKE2 Service User Account"
+	chown -R "${INSTALL_RKE2_USER}":"${INSTALL_RKE2_USER}" "$(dirname ${BASE_DIR})"
```
Comment (galal-hussein) [inline install.sh:849]: nit: indentation

Comment (briandowns) [reply install.sh:849]: Ugh. Good catch!

**Pull request merged**